src/wp-includes/class-wp-user-query.php

```diff
@@ -790,6 +793,11 @@ public function query() {
 
 		$qv =& $this->query_vars;
 
+		// Do not cache results if more than 3 fields are requested.
+		if ( is_array( $qv['fields'] ) && count( $qv['fields'] ) > 3 ) {
```

Comment:
I understand the reasoning behind this but it might be wise to leave it as is.

If the files value is modified by filters (either those that exist now or added in the future) then it's best to make sure the condition is up to date.

Reply:
Thanks @peterwilsoncc Make sanse.